### DISCUSSION TITLE
Info: Installation from source for Ubuntu 20.04 users, missing dependency gir1.2-handy-1

### DISCUSSION THREAD

**Discussion by michael-schaefer-eu** [pdfarranger/pdfarranger#651] [Q&A] [answered]
Installation from source for Ubuntu 20.04, missing dependency gir1.2-handy-1
----------------

If you try to [install from source](https://github.com/pdfarranger/pdfarranger#install-from-source) on Ubuntu 20.04, errors for `gir1.2-handy-1` will occur.

To solve this, you need to add the PPA for package `gir1.2-handy-1` prior the installation:

```
sudo add-apt-repository ppa:apandada1/libhandy-1
sudo apt-get update
sudo apt-get install python3-pip python3-distutils-extra python3-wheel python3-gi python3-gi-cairo gir1.2-gtk-3.0 gir1.2-poppler-0.18 gir1.2-handy-1 python3-setuptools
pip3 install --user --upgrade https://github.com/pdfarranger/pdfarranger/zipball/main
```

**Comment by dreua** [comment]
Thanks, I believe we need to update the Readme to work with #621

**Comment by jeromerobert** [comment]
I don't like the idea to add that particular PPA to our README. Depending on how this PPA will be maintained Google will remain the most reliable source. May be the package from Hirsute would also work. Anyway the next LTS is due in a month.

I'll also won't any note on how to install it in Debian Buster with backports either.

**Comment by dreua** [comment]
Sorry didn't spot the PPA part there. Does that mean #621 will not be compatible with older systems? Maybe we should just revert it until the library is commonly available.

**Comment by michael-schaefer-eu** [comment]
Sorry, the `bug` label was my mistake, this issue was meant as an **information** for users of Ubuntu 20.04 as I couldn't find it here but had to search for it and because Ubuntu 20.04 has only `gir1.2-handy-0` in it's standard repos.

Also Ubuntu 20.04 is [supported](https://ubuntu.com/about/release-cycle) until 2030 and it would install the version 1.4.x of pdfarranger, I thought this info might be helpful for some users in the future that want the newest version :)

**Comment by jeromerobert** [comment]
My first issue converted to a *Github Discussion* ;-)

I think it should be possible make LibHandy optional.

@oscfdezdz So far we (at least kbengs and I) have not been able to see the benefit of LibHandy. On my desktop it actually show no visual differences. If I want to work on making it optional I need to see those differences. Could you please tell us what themes or settings are needed to see the differences ? Could you also provide screenshots of "with LibHandy" and "without LibHandy".

@oscfdezdz We are going to put #649 on hold until LibHandy has been made optional

**Comment by oscfdezdz** [comment]
Not sure about WIndows, but for now the only difference that can be seen is the rounded bottom corners at least in (default) GNOME Shell, other DEs may or may not show them I guess.

With LibHandy:
![Captura desde 2022-03-29 09-24-22](https://user-images.githubusercontent.com/42654671/160568798-e1a9d96d-588e-4f61-a63f-38ace5554489.png)

Without LibHandy:
![Captura desde 2022-03-29 09-27-59](https://user-images.githubusercontent.com/42654671/160568803-a6e411ce-eba9-4d35-83d3-25d372367122.png)

#621 was just an initialization, it did not make any other notable changes, although some widgets/dialogs like "Edit Properties" could benefit from classes like `Handy.PreferencesWindow`, for example. 

I have tried dropping #621 and #649 would work without it if that makes things easier for making LibHandy optional, but it would still need to make sure version 1.6 is available.

**Comment by jeromerobert** [comment, marked as answer]
Handy is now optional (#655)
